flink/v1.16/flink/src/main/java/org/apache/iceberg/flink/source/ScanContext.java

```diff
@@ -125,7 +137,24 @@ private void validate() {
             startSnapshotId == null,
             "Invalid starting snapshot id for SPECIFIC_START_SNAPSHOT_ID strategy: not null");
       }
+
+      Preconditions.checkArgument(
+          branch == null,
+          String.format(
+              "Cannot scan table using ref %s configured for streaming reader yet", branch));
+
+      Preconditions.checkArgument(
+          tag == null,
+          String.format("Cannot scan table using ref %s configured for streaming reader", tag));
```

Comment:
#5984 seems to be the prerequisite for flink to implement stream incremental read.